buddy_alloc.h

```diff
@@ -115,15 +119,54 @@ void buddy_set_left_bias(struct buddy *buddy);
  */
 void buddy_set_optimal_fit(struct buddy *buddy);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif /* BUDDY_ALLOC_H */
 
 #ifdef BUDDY_ALLOC_IMPLEMENTATION
 #undef BUDDY_ALLOC_IMPLEMENTATION
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #ifndef BUDDY_ALLOC_ALIGN
 #define BUDDY_ALLOC_ALIGN (sizeof(size_t) * CHAR_BIT)
 #endif
 
+#ifdef _MSC_VER
+    
+/* Aligns value using provided mask
+ *  x    - value to align
+ *  mask - desired alignment minus one (2^n - 1)
+ */
+#define BUDDY_ALIGNUP(x, mask) ((x) + (-(x) & mask))
+
+#ifndef _SSIZE_T_DEFINED
+typedef signed long long ssize_t;
+#define _SSIZE_T_DEFINED
+#endif
+
+#ifndef __alignof__
+#define __alignof__(x) _Alignof(x)
+#endif
+
+/* Returns desired alignment for a single element */
+static size_t buddy_alignof( size_t what )
+{
+    // TO-DO
+    ((void*)what);
+    return CHAR_BIT;
```

Comment:
Please read comment on line 415

Reply:
This will work although my reason for having `BUDDY_ALLOC_ALIGN` as a macro was to allow users to define it themselves to a different value.